common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -883,6 +918,25 @@ def answer_available(self):
 
         return False
 
+    def correctness_available(self):
+        """
+        Is the user allowed to see whether she's answered correctly?
+
+        Limits access to the correct/incorrect flags, messages, and problem score.
+        """
+        if self.show_correctness == SHOW_CORRECTNESS.NEVER:
+            return False
+        elif self.runtime.user_is_staff:
+            # This is after the 'never' check because admins can see correctness
```

Comment:
@cahrens @srpearce 

I agree it's totally unintuitive, and we had a client just last week with this same misunderstanding about `SHOWANSWER`.

> I didn't know that you could set a coursewide default for SHOWANSWER in the course advanced settings (though I checked Studio and lo and behold it's true!), and we don't document that anywhere that I can find.

It's available in the help text on the Problem Settings dialog, but I couldn't find any documentation either:

![screen shot 2017-04-11 at 1 49 55 pm](https://cloud.githubusercontent.com/assets/7556571/24892902/2024c930-1ebf-11e7-8c29-03e482718d01.png)

> I finally saw/understood your other comment about this, and see the similar behavior in `answer_available`. I am OK with this remaining as-is for consistency.

Thanks, I'm glad that consistency is acceptable.

I've [updated the docs PR](https://github.com/edx/edx-documentation/pull/1466/commits/264b2cfa5e4e5891d1994c49bd3ac44a34055014) to remind course authors to view the problems in Student View to get a true picture of what's being shown to their students.